### PR TITLE
Scheduled Command Management

### PR DESCRIPTION
**Pull request by grebneerg**
Adds `gort schedule get` and `gort schedule delete` to get a list of currently scheduled commands and remove scheduled commands by id, respectively. These are exposed through the default bundle as `gort:schedule-list` and `gort:cancel`.

Additionally, schedules are now stored in the data storage, and thus persist across gort restarts if you are using postgres. For the moment, an `Original` field has been added to `command.Command` to enable easy serialization, but this leads to some unfortunate fragility around where it is set and where it is used for parsing.

One thing slightly sub-par here is terminology. I've been stuck between the ideas of referring to _commands_ that are scheduled and cancelled vs referring to _schedules_ that are created and deleted. Any preferences between those two conceptual models, so that I can make the naming more consistent?

One other note is that unlike many of the other subcommands in the default bundle, the schedule subcommand is exposed as three gort commands instead of just one with further subcommands as parameters. This seems more natural given the current command parsing limitations (especially with regards to flag parsing, see #225), but is less clean overall. Of course, exposing three separate commands also allows me to specify custom help text for each. The help text would otherwise just be the default for each subcommand, which assumes that they are being called from the command line, and thus may be incongruous with the command's actual use.

### CODE REVIEW

File: cli/schedule-delete.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+)
+
+const (
+	schedulDeleteUse    = "delete"
+	scheduleDeleteShort = "Deletes a scheduled command"
+	scheduleDeleteLong  = "Deletes a scheduled command."
+	scheduleDeleteUsage = `Usage:
+gort schedule delete [flags] id
+
+id is the id of the scheduled command to delete.
```
Comment (clockworksoul) [inline cli/schedule-delete.go:37]: This should also describe what the command does.

File: cli/schedule-get.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+)
+
+const (
+	scheduleGetUse   = "get"
+	scheduleGetShort = "Get a list of schedules commands"
+	scheduleGetLong  = "Get a list of schedules commands."
+	scheduleGetUsage = `Usage:
+gort schedule get [flags]
+
+Flags:
```
Comment (clockworksoul) [inline cli/schedule-get.go]: What does the command do?

File: cli/schedule-get.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	if err != nil {
+		return err
+	}
+
+	info, err := c.SchedulesGet()
+	if err != nil {
+		return err
+	}
+
+	maxId, maxAdapter, maxChannel, maxCron := 2, 7, 7, 4
```
Comment (clockworksoul) [inline cli/schedule-get.go]: Should these be constants?
Also, can you comment them to make it clearer what they do and maybe why they're the values they are?

File: scheduler/scheduler.go [hunk 4]
```diff
@@ -112,21 +121,89 @@ func Schedule(ctx context.Context, cmd data.ScheduledCommand) error {
...
+
+// Cancel cancels the scheduled command with the given id. Even in the event of
+// an error, the command may still be cancelled.
+func Cancel(ctx context.Context, scheduleID int64) error {
+	da, err := dataaccess.Get()
+	if err != nil {
+		return err
+	}
+
+	err = da.ScheduleDelete(ctx, scheduleID)
```
Comment (clockworksoul) [inline scheduler/scheduler.go:171]: Idiomatically I'd expect to see something more like:

```go
        err = da.ScheduleDelete(ctx, scheduleID)
	if err != nil {
		return err
	}

	err = cron.RemoveByTag(fmt.Sprintf("%d", scheduleID))
	if err2 != nil {
		return err
	}
```

**Review by clockworksoul — COMMENTED** [review]
Just a couple of comments for such a large PR!

File: cli/schedule-get.go (hunk 3 above)
Comment (grebneerg) [reply cli/schedule-get.go]: Definitely not constants, I use them to properly space my table.

File: scheduler/scheduler.go (hunk 4 above)
Comment (grebneerg) [reply scheduler/scheduler.go:171]: I was thinking that we might still want to end the job even if it fails to be removed from `da`, but maybe creating that kind of inconsistency is worse.

**Pull request merged**